db/model/generator.js

```diff
@@ -373,5 +378,48 @@ module.exports = function generator(seq, dataTypes) {
     },
     paranoid: true,
   });
+
+  /*
+   * Validate Subject Query
+   *
+   * @param {String} subjectQuery string
+   * @return {String} returns Subject Query or throws an error
+   *                          if subjectQuery fails with validation
+   *                          rules
+   */
+  Generator.validateSubjectQuery = function (subjectQuery) {
+    // subjectQuery should start with '?'
+    if (subjectQuery.charAt(0) !== '?') {
+      throw new ValidationError('subjectQuery ValidationError',
+        'subjectQuery must start with "?"');
+    }
+
+    // size of subjectQuery should be greater than 6
+    if (subjectQuery.length <= 6) {
+      throw new ValidationError('subjectQuery ValidationError',
+        'subjectQuery must be longer than 6 characters');
+    }
+
+    const subjectQueryString = subjectQuery.substr(1);
+
+    const splitSubjectQuery = subjectQueryString.split('&');
+
+    splitSubjectQuery.forEach((sq) => {
+      const splitSQ = sq.split('=');
+
+      if (splitSQ.length !== 2) {
+        throw new ValidationError('subjectQuery ValidationError',
+          'Format of subjectQuery must be "?{key}={value}"');
+      }
+
+      if (splitSQ[0] == 'tags' && splitSQ[1].indexOf('*') > -1) {
```

Comment:
Wondering if we also decided to prevent queries like absolutePath=* and name=* . I think we talked about it but I could not remember what we decided on.

Reply:
Since "absolutePath=*" (or "name=*") are the same as not including that clause at all, we should treat it the same as if it wasn't there at all, and send back validation error if there are no other query params